Rationale.md

```diff
@@ -13,10 +13,17 @@ codebases, we'll revisit the alternatives listed below, reevaluate the tradeoffs
 and update the [design](AstSemantics.md) before the MVP is finalized.
 
 
-## Why AST?
+## Why a stack machine?
 
-Why not a register- or SSA-based bytecode?
-* Trees allow a smaller binary encoding: [JSZap][], [Slim Binaries][].
+Why not an AST, or a register- or SSA-based bytecode?
+
+* We started with an AST and generalized to a [structured stack machine](AstSemantics.md). ASTs allow a 
+  dense encoding and efficient decoding, compilation, and interpretation.
+  The stack machine is a generalization of ASTs allowed in previous versions while allowing
```

Comment:
Maybe good to specify "The _structured_ stack machine _of WebAssembly_ is a ...".